languages/elixir/concepts/maps/about.md

```diff
@@ -1 +1,86 @@
-TODO: add information on maps concept
+[Maps][maps] are a data structure that holds key-value pairs.
+
+- Keys can be of any type, but must be unique.
+- Values can be of any type, they do not have to be unique.
+- Maps do not guarantee the order of their contents despite appearing to do so.
+  - Their underlying implementation gives this misperception:
+    - At small sized (<=32 entries), they are implemented as an ordered [Keyword list][keyword-list]
+    - At larger sizes (>32 entries), they are implemented as a [hash array mapped trie][hamt] &#91;[1][stackoverflow]&#93;
+- Maps can be declared with a literal form:
+
+  ```elixir
+  # An empty map
+  %{}
+
+  # A map with the atom key :a associated to the integer value 1
+  %{a: 1}
+
+  # A map with the string key "a" associated to the float value 2.0
+  %{"a" => 2.0}
+
+  # A map with the map key %{} with the list value [1,2,3]
+  %{%{} => [1,2,3]}
+  ```
+
+- Maps can also be instantiated using [`Map.new`][map-new] from the [Map module][map-module]. This might be used if you already have an enumerable collection to turn into a list:
+
+  ```elixir
+  kw_list = [a: 1, b: 2]
+  Map.new(kw_list)
+  # => %{a: 1, b: 2}
+  ```
+
+- Values in a map can be accessed in many different way:
+
+  ```elixir
+  my_map = %{key: "value"}
+
+  # with a dot if the key is an atom
+  my_map.key
+  # => "value"
+
+  # with [], a syntax provided by the Access behaviour
+  my_map[:key]
+  # => "value"
+
+  # with pattern matching
+  %{key: x} = my_map
+  x
+  # => "value"
+
+  # with Map.get/2
+  Map.get(my_map, :key)
+  # => "value"
+  ```
```

Comment:
I added this snippet about accessing values and modified the code example below to use Map.delete instead of Map.get.